### ISSUE TITLE
COMPRESSION_ERROR Warning on all requests.

### ISSUE THREAD

**Issue by andrewpatrickburns** [loadimpact/k6#897]
Hi,

I started using k6 today to load test my REST API, I currently have 3 versions of it running, two instances on linux VMs, and another instance in a local tomcat instance, running on windows.

When I use k6 to load test my local instance of the API, everything seems fine.

When I use k6 to load test my linux VM instances, I get warnings on each request that the request has failed, like so:

![image](https://user-images.githubusercontent.com/16542095/51324494-c5410280-1a62-11e9-8294-96c296ab9172.png)

I am receiving the data, as shown in the image with the `data_received` property.

I'm using a pretty simple setup at the moment:

```
let headers = {
    authorization: "Bearer " + authToken,
    contentType: "application/json"
}

export let options = {
    insecureSkipTLSVerify: true,
    vus: 5,
    duration: "120s"
  };

export default function() {
  let res = http.get(baseUrl + path, { headers: headers });

  check(res, {
    "status was 200": (r) => r.status == 200,
    "transaction time OK": (r) => r.timings.duration < 5000
  });

  sleep(1);

  res = http.get(baseUrl + path, { headers: headers });

  check(res, {
    "status was 200": (r) => r.status == 200,
    "transaction time OK": (r) => r.timings.duration < 5000
  });

};
```

Am I doing something wrong?

**Comment by MStoykov** [comment]
This sounds like k6 and the server you are using are having trouble communicating for some reason. Golang (the language k6 is written in) has some compatibility issues https://github.com/golang/go/issues/25023 . Most of those seem to be fixed, especially the ones mentioning `COMPRESSION_ERROR`.

Can you check that first you are running v0.23.1  of k6 and that it was compiled with golang 1.11.x (if you compiled it yourself). 

If you are using the correct version can you tell us what server and version you are using ? Maybe either it or golang doesn't implement something correctly and we will need to test it to see. Maybe there is a proxy in between ?

**Comment by andrewpatrickburns** [comment]
@MStoykov Thanks for your reply, I'm running `Apache Tomcat v8.5.31` and `k6 v0.23.1`. There is no proxy between k6 and the server.

I'm currently using Windows to run k6, installed via the .msi available on the website.

**Comment by MStoykov** [comment]
Hm this to me looks like a Tomcat bug. They have fixed something similar in [coyote in  v8.5.10 ](https://tomcat.apache.org/tomcat-8.5-doc/changelog.html) (search for "compression error"). I am going to test it but it might take me some time to get to this and probably even more time until I setup something.  

Can you try to fidget with your tomcat configuration. Given that it is only Linux, you have problems on and [this stackoverflow question](https://stackoverflow.com/questions/50432847/tomcat-8-5-29-http-2-is-not-supporting-gzip-compression) I am guessing maybe this is something to do with `sendfile`. Can you try to disable compression or `sendfile` and see if this changes anything?

**Comment by na--** [comment]
I'm closing this, since there hasn't been any updates for some time, and the most likely explanation seemed to be something on the tomcat side. Feel free to comment or reopen if that isn't the case or if anyone else hits the same issue, but under different circumstances.
